src/connection.ts

```diff
@@ -9,7 +9,10 @@ function getProvider(config: any): Provider {
     switch (config.type) {
         case undefined:
             return config;
-        case 'JsonRpcProvider': return new JsonRpcProvider(config.args.url);
+        case 'JsonRpcProvider': return new JsonRpcProvider({
```

Comment:
1. Can we configure this with the entire `config.args` object rather than have to manually translate sub-keys under that object?  In the future it will be less code if we can add new properties to the `config.args` type and know that they are passed 1:1 into the Provider constructor
2.  It would be nice if consumers of this object could add any arbitrary headers they want, without us needing to be involved in adding code cases to support all of them explicitly.  e.g. something like `const { url, headers } = config.args;`.
We could still document that `headers` can contain any number of optional keys, but it would be really nice on the consumer side if the shape of `headers` here would be a 1:1 match with the _actual header keys to be sent to the RPC node_; that way anyone creating a provider instance could intuitively configure their headers without needing to identify which keys in `args` correspond to which keys in `connection.headers`

Reply:
Agree, now instead of the key itself we will pass the headers object. (example included)